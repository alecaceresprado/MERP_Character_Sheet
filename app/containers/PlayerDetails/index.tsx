import * as React from 'react';
import { useParams } from 'react-router-dom';

export default function PlayerDetails() {
  const { id } = useParams();

  return <div>Player ID: {id}</div>;
}
