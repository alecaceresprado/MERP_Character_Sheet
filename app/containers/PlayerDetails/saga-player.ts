import { call, put, takeLatest } from 'redux-saga/effects';

import { AppActions, AppActionTypes } from 'state/app/app.actions';
import request from 'utils/request';

export function* getPlayersDetail(action) {
    const requestURL = `/mockPlayer/${action.payload}`;

    try {
        const playerDetails = yield call(request, requestURL);
        yield put(AppActions.playerDetailsLoaded(playerDetails));
    } catch (err) {
        yield put(AppActions.playerDetailsLoadingError(err));
    }
}

export default function* getPlayerDetailData() {
    yield takeLatest(AppActionTypes.LOAD_PLAYER_DETAIL, getPlayersDetail);
}
