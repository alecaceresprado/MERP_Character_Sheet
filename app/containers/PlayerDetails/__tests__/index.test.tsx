import React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import 'jest-styled-components';

import PlayerDetails from '../index';

const renderer = createRenderer();

jest.mock('react-router-dom', () => ({
  useParams: () => ({ id: '1' })
}));

describe('<PlayerDetails />', () => {
  it('should render and match the snapshot', () => {
    renderer.render(<PlayerDetails />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
