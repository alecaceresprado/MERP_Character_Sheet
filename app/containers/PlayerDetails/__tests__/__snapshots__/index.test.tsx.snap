// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlayerDetails /> should render and match the snapshot 1`] = `
<div>
  Player ID: 
  1
</div>
`;
